assembly/src/assembler/mod.rs

```diff
@@ -59,13 +149,13 @@ impl Assembler {
         self
     }
 
-    /// Adds the specified [ModuleProvider] to the assembler.
-    pub fn with_module_provider<P>(mut self, provider: P) -> Self
+    /// Adds the library to provide modules for the compilation.
+    pub fn with_library<L>(mut self, library: L) -> Result<Self, AssemblyError>
```

Comment:
We need to have a way to instantiate an assembler with several libraries. We can use this to do that, but I wonder if it would be awkward in certain situations. For example, let's say we have a vector of libraries, how would we create an assembler from that?

An alternative could be something like `with_libraries()` which takes an iterator as an input. Maybe we should have both `with_library()` and `with_libraries()`? (though, the names are quite close to each other).

Reply:
That makes sense and could be a convenient helper, especially for the library bundle that we are creating in the other issue. Added!